entc/gen/func.go

```diff
@@ -127,18 +124,31 @@ func pascal(s string) string {
 	return strings.Join(words, "")
 }
 
-// camel converts the given column name into a camelCase.
+// pascal converts the given name into a PascalCase.
+//
+//	user_info => UserInfo
+//	full_name => FullName
+//	user_id   => UserID
+//	full-admin   => FullAdmin
+//
+func pascal(s string) string {
+	words := strings.FieldsFunc(s, isSeparatorRune)
+	return pascalWords(words)
+}
+
+// camel converts the given name into a camelCase.
 //
-//	user_info => userInfo
-//	full_name => fullName
-//	user_id   => userID
+//	user_info  => userInfo
+//	full_name  => fullName
+//	user_id    => userID
+//	full-admin => fullAdmin
 //
 func camel(s string) string {
-	words := strings.SplitN(s, "_", 2)
+	words := strings.FieldsFunc(s, isSeparatorRune)
```

Comment:
Sadly, `strings.FieldsFunc` doesn't have a `SplitN` equivalent so I've split out the `pascalWords` function which assumes the string is already split.